service/controller/app/v1/resource/chart/create.go

```diff
@@ -0,0 +1,28 @@
+package chart
+
+import (
+	"context"
+	"fmt"
+	"github.com/giantswarm/app-operator/service/controller/app/v1/key"
+	"github.com/giantswarm/microerror"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+	customResource, err := key.ToCustomResource(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	name := key.AppName(customResource)
+	chart, err := r.g8sClient.ApplicationV1alpha1().Charts(r.watchNamespace).Get(name, v1.GetOptions{})
```

Comment:
We can't always use the g8sClient. For App Catalog we need to connect to the tenant cluster. Its merged now so you can use the kubeconfig service to get the right client.

https://github.com/giantswarm/app-operator/blob/873ea9601aa18a42bbe35947cf4e3a953af95b6a/service/controller/app/v1/kubeconfig/kubeconfig.go#L72